client/src/actions/index.js

```diff
@@ -28,3 +34,34 @@ export const fetchProfile = profileId => async dispatch => {
     console.log("fetchProfile: ", err);
   }
 };
+
+export const addUpdate = (data, callback) => async dispatch => {
+  try {
+    if (data.image_url) {
+      callback();
+      dispatch({ type: LOADING_UPDATE, payload: true });
+      const formData = new FormData();
+      formData.append("image", data.image_url);
+      const config = {
+        headers: {
+          Authorization: "Client-ID b981e83d44eafce"
+        }
+      };
+      const imgrData = await axios.post(
+        "https://api.imgur.com/3/image",
+        formData,
+        config
+      );
+      data.image_url = imgrData.data.data.link;
+    } else {
+      data.image_url = null;
+    }
+    const updateData = await axios.post("/api/addUpdate", { ...data });
+    dispatch({ type: ADD_UPDATE, payload: updateData.data });
+    if (!data.image_url) {
+      callback();
+    }
```

Comment:
if no picture then wait until new post ready before redirecting to posts page, to prevent loader flashing quickly